### PR TITLE
fix goLevelDBIterator

### PR DESCRIPTION
**Pull request by melekes**
remove unnecessary return statement

### CODE REVIEW

**Comment by melekes** [comment]
@odeke-em see https://github.com/tendermint/tmlibs/pull/62/commits/0b22b27bbb08947e3875dcfe22f95b983635eb0a . I updated the commit's description. Hope it's more clear right now.

File: db/go_level_db.go [hunk 1]
```diff
@@ -118,12 +118,12 @@ func (db *GoLevelDB) Stats() map[string]string {
...
 
 type goLevelDBIterator struct {
-	iterator.Iterator
+	source iterator.Iterator
 }
 
 // Key returns a copy of the current key.
 func (it *goLevelDBIterator) Key() []byte {
-	key := it.Key()
+	key := it.source.Key()
```
Comment (ebuchman) [inline db/go_level_db.go]: oy can't believe we missed this first time around :(

**Comment by ebuchman** [comment]
@odeke-em we had infinite loops before, so need a way to name the inner iterator. Could just refer to it as `.Iterator`, but an explicit name is great too. LGTM

File: db/go_level_db.go (hunk 1 above)
Comment (cloudhead) [reply db/go_level_db.go]: I know right :/

**Pull request merged**